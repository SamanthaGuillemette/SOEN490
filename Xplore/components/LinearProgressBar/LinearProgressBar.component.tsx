import { StyleProp, ViewStyle } from "react-native";
import { ProgressBar } from "react-native-paper";
import { colors } from "../../constants";
import { useThemeColor } from "../../hooks";
import { View } from "../View";
import styles from "./LinearProgressBar.styles";

interface LinearProgressBarProps {
  color?: keyof typeof colors.light & keyof typeof colors.dark;
  progress: number; // Range 0 - 100  ==> Example: 30, 50, 80, etc.
  style?: StyleProp<ViewStyle>;
}

export const LinearProgressBar = (props: LinearProgressBarProps) => {
  const { color = "primary", progress, style } = props;
  const customProgressColor = useThemeColor(color);
  const background = useThemeColor("background");

  return (
    <View>
      <ProgressBar
        progress={progress / 100}
        color={customProgressColor}
        style={[
          style,
          styles.progressBarMain,
          { backgroundColor: background, borderColor: background },
        ]}
      />
    </View>
  );
};
