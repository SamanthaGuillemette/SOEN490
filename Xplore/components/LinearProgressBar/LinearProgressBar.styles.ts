import { StyleSheet } from "react-native";

export default StyleSheet.create({
  progressBarMain: {
    borderRadius: 8,
    borderWidth: 2,
    height: 8,
  },
});
