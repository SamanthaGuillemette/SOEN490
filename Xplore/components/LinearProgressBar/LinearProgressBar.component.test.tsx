import React from "react";
import { cleanup } from "react-native-testing-library";
import { LinearProgressBar } from "./LinearProgressBar.component";
import renderer from "react-test-renderer";

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

afterAll(() => {
  cleanup();
});

describe("LinearProgressBar should render", () => {
  it("should render the progress bar", () => {
    const bar = renderer.create(<LinearProgressBar progress={0.2} />);
    expect(bar).toMatchSnapshot();
  });

  it("should render the progress bar with a different color", () => {
    const bar = renderer.create(
      <LinearProgressBar progress={0.2} color={"success"} />
    );
    expect(bar).toMatchSnapshot();
  });
});
