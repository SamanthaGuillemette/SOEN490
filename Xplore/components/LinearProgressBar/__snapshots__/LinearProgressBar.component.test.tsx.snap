// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearProgressBar should render should render the progress bar 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      Object {
        "busy": true,
      }
    }
    accessibilityValue={
      Object {
        "max": 100,
        "min": 0,
        "now": 0.2,
      }
    }
    accessible={true}
    onLayout={[Function]}
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "borderColor": "#F5F5F5",
          "borderRadius": 8,
          "borderWidth": 2,
          "height": 8,
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    />
  </View>
</View>
`;
